### PR TITLE
hotfix：fix test failure in cuda environment

### LINKED ISSUE #286: RuntimeError: CUDA error: an illegal memory access was encountered
Hello,

When running some of the test code, I met the following error:
```python
% python test/discrete/test_sac.py                                                                                                                               (git)-[master]-
tianshou/data/buffer.py:348: UserWarning: ListReplayBuffer will be replaced in version 0.4.0.
  warnings.warn("ListReplayBuffer will be replaced in version 0.4.0.")
Epoch #1:   0%|                                                                                                                                         | 0/1000 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "test/discrete/test_sac.py", line 121, in <module>
    test_discrete_sac()
  File "test/discrete/test_sac.py", line 104, in test_discrete_sac
    result = offpolicy_trainer(
  File "tianshou/trainer/offpolicy.py", line 92, in offpolicy_trainer
    result = train_collector.collect(n_step=collect_per_step)
  File "tianshou/data/collector.py", line 270, in collect
    self.data.act = to_numpy(result.act)
  File "tianshou/data/utils/converter.py", line 18, in to_numpy
    return x.detach().cpu().numpy()
RuntimeError: CUDA error: an illegal memory access was encountered
```

The same error shows for `test/discrete/test_ppo.py`. However, it works well for some other RL algorithms, such as DQN (`test/discrete/test_dqn.py`).

Here is my environment:
- tianshou: 0.3.1 (commit  f0129f4ca774e89f0def2c7ec0d14791ab98f22e)
- pytorch: v1.6.0
- python: v3.8.5
- OS: Ubuntu 16.04

### PR DESCRIPTION
**Pull request by ChenDRAG**
For now, if you run
`pytest test --cov tianshou -s --durations 0 -v` on a gpu mechine, it will raise error(due to cuda/cpu data conversion)
This pr fix that.

### CODE REVIEW

File: tianshou/utils/net/continuous.py [hunk 1]
```diff
@@ -49,7 +49,8 @@ def __init__(
         self.output_dim = np.prod(action_shape)
         input_dim = getattr(preprocess_net, "output_dim",
                             preprocess_net_output_dim)
-        self.last = MLP(input_dim, self.output_dim, hidden_sizes)
+        self.last = MLP(input_dim, self.output_dim,
+                        hidden_sizes, device = self.device)
```
Comment (Trinkle23897) [inline tianshou/utils/net/continuous.py]:
```suggestion
                        hidden_sizes, device=self.device)
```

File: tianshou/utils/net/continuous.py [hunk 2]
```diff
@@ -98,7 +99,7 @@ def __init__(
         self.output_dim = 1
         input_dim = getattr(preprocess_net, "output_dim",
                             preprocess_net_output_dim)
-        self.last = MLP(input_dim, 1, hidden_sizes)
+        self.last = MLP(input_dim, 1, hidden_sizes, device = self.device)
```
Comment (Trinkle23897) [inline tianshou/utils/net/continuous.py]:
```suggestion
        self.last = MLP(input_dim, 1, hidden_sizes, device=self.device)
```

File: tianshou/utils/net/continuous.py [hunk 3]
```diff
@@ -164,10 +165,12 @@ def __init__(
         self.output_dim = np.prod(action_shape)
         input_dim = getattr(preprocess_net, "output_dim",
                             preprocess_net_output_dim)
-        self.mu = MLP(input_dim, self.output_dim, hidden_sizes)
+        self.mu = MLP(input_dim, self.output_dim,
+                      hidden_sizes, device = self.device)
```
Comment (Trinkle23897) [inline tianshou/utils/net/continuous.py]:
```suggestion
                      hidden_sizes, device=self.device)
```

File: tianshou/utils/net/continuous.py [hunk 4]
```diff
@@ -164,10 +165,12 @@ def __init__(
...
         self._c_sigma = conditioned_sigma
         if conditioned_sigma:
-            self.sigma = MLP(input_dim, self.output_dim, hidden_sizes)
+            self.sigma = MLP(input_dim, self.output_dim,
+                             hidden_sizes, device = self.device)
```
Comment (Trinkle23897) [inline tianshou/utils/net/continuous.py]:
```suggestion
                             hidden_sizes, device=self.device)
```

**Review by Trinkle23897 — DISMISSED** [review]

**Comment by ChenDRAG** [comment]
finished I think, ready for review

**Review by Trinkle23897 — APPROVED** [review]

**Pull request merged**